lib/src/logger.dart

```diff
@@ -27,21 +27,29 @@ class Logger {
 
   /// The full name of this logger, which includes the parent's full name.
   String get fullName =>
-      (parent == null || parent.name == '') ? name : '${parent.fullName}.$name';
+      parent?.name.isNotEmpty == true ? '${parent!.fullName}.$name' : name;
```

Comment:
Was this a bug before? I'm surprised that we check `parent.name` instead of `parent.fullName`, but then use `fullName` in the string...

Reply:
This is fine - the empty named logger is always the root logger no matter what afaik, and name is cheaper to check.